.github/workflows/python-pr.yml

```diff
@@ -12,13 +12,13 @@ concurrency:
 
 jobs:
   test:
-    name: Quick Test
-    runs-on: self-hosted
-
+    name: Multi-OS Tests
+    runs-on: ${{ matrix.os }}
     strategy:
-      fail-fast: true
+      fail-fast: false
```

Comment:
Fail fast should remain true. The same error is likely to occur on the other runs, and that just means wasted computational resources.